### PR TITLE
[core] Support create table column in-sensitive in HiveCatalog

### PR DESCRIPTION
**Pull request by xuzifu666**
### Purpose
Some time user need migrate their sql from hive to paimon during spark or flink job migrate，but their sql combine with lowercase and uppercase，so the pr is aim to support it.
when we create table with uppercase column
before fix:

after fix:

Linked issue: https://github.com/apache/paimon/issues/3823

### CODE REVIEW

File: paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkGenericCatalog.java [hunk 1]
```diff
@@ -284,6 +288,13 @@ private CaseInsensitiveStringMap autoFillConfigurations(
         Map<String, String> newOptions = new HashMap<>(options.asCaseSensitiveMap());
         fillAliyunConfigurations(newOptions, hadoopConf);
         fillCommonConfigurations(newOptions, sqlConf);
+
+        // add case-insensitive from sql conf
+        Option<String> insensitive = sqlConf.getAllConfs().get(sparkCaseInsensitive);
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkGenericCatalog.java:293]: `sqlConf.caseSensitiveAnalysis`?

Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkGenericCatalog.java:293]: Also modify `SparkCatalog`?

Comment (xuzifu666) [reply paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkGenericCatalog.java:293]: OK，sqlConf.caseSensitiveAnalysis is better.

File: paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkGenericCatalog.java [hunk 2]
```diff
@@ -85,6 +87,8 @@ public class SparkGenericCatalog extends SparkBaseCatalog implements CatalogExte
 
     private CatalogPlugin sessionCatalog = null;
 
+    private final String sparkCaseInsensitive = "spark.case-insensitive";
```
Comment (Zouxxyy) [inline paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkGenericCatalog.java:90]: If `case-insensitive` is a catalog conf, it is recommended to set like `spark.sql.catalog.$catalogName.case-insensitive` to use it

Comment (xuzifu666) [reply paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkGenericCatalog.java:90]: Thanks，would change like @JingsongLi  suggession.

File: paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java [hunk 3]
```diff
@@ -580,7 +581,7 @@ private void alterTableToHms(Table table, Identifier identifier, TableSchema new
 
     @Override
     public boolean caseSensitive() {
-        return false;
+        return catalogOptions.get(CASE_INSENSITIVE);
```
Comment (JingsongLi) [inline paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java:584]: CASE_INSENSITIVE caseSensitive?

File: paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalogOptions.java [hunk 4]
```diff
@@ -85,5 +85,12 @@ public final class HiveCatalogOptions {
                                                             + "E.g. specifying \"conf:a.b.c\" will add \"a.b.c\" to the key, and so that configurations with different default catalog wouldn't share the same client pool. Multiple conf elements can be specified."))
                                     .build());
 
+    public static final ConfigOption<Boolean> CASE_INSENSITIVE =
```
Comment (JingsongLi) [inline paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalogOptions.java:88]: can this be a option in CatalogOptions?

Comment (xuzifu666) [reply paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalogOptions.java]: Done，change it to CatalogOptions

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**